tests/xharness/BCLTestImporter/Templates/IProjectFilter.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Xharness.BCLTestImporter.Templates {
+
+	/// <summary>
+	/// There are cases in which projects are ignored in certain platforms, either because
+	/// we do know we have issues in the plafrom or because certain dlls are not present. The interface allows
+	/// to ignore projects and dlls per platform. Template objects should honour this filter.
+	/// </summary>
+	public interface IProjectFilter {
+
+		/// <summary>
+		/// True if a specific dlls should be ignored from a project in a platform.
+		/// </summary>
+		/// <param name="platform">The platform whose project is going to be generated.</param>
+		/// <param name="assembly">The assembly that is part of the test project.</param>
+		/// <returns>True if the dll was to be excluded, false otherwhise.</returns>
+		bool ExcludeDll (Platform platform, string assembly);
```

Comment:
>ExcludeDll [](start = 7, length = 10)

I think the naming here might be better when something like `ShouldExclude` or `ShouldBeExcluded` or `IsDllExcluded`...
`Exclude` only hints that there will be a side effect of some sort.

But no hard feelings about this, up to you.